Docs/reference/content/reference/driver/crud/reading.md

```diff
@@ -312,4 +312,122 @@ await OutAsync("myNewCollection");
 ```
 ```json
 { $out: 'myNewCollection' }
-```
\ No newline at end of file
+```
+
+### Text Search
+[Atlas Search](https://docs.atlas.mongodb.com/atlas-search) makes it easy to build fast, relevance-based search capabilities on top of your MongoDB data. Try it today on [MongoDB Atlas](https://www.mongodb.com/cloud/atlas), our fully managed database as a service.
+
+The following example can be used alongside the [Atlas Search Tutorial](https://docs.atlas.mongodb.com/reference/atlas-search/tutorial/)
+
+```csharp
+var client = new MongoClient(connectionString);
+var collection = client.GetDatabase("sample_mflix").GetCollection<BsonDocument>("movies");
+var searchStage = BsonDocument.Parse(@"
+{
+	$search: {
+    	'text': {
+    	    'query': 'baseball',
+            'path': 'plot'
+        }
+    }
+}");
+PipelineDefinition<BsonDocument, BsonDocument> pipeline = new []
+{
+    searchStage,
+    new BsonDocument("$limit", 5),
+    new BsonDocument
+    {
+        {"$project", new BsonDocument{ {"_id", 0}, {"title", 1}, {"plot", 1} }}
+    }
+};
+var moviesAboutBaseball = collection.Aggregate(pipeline).ToList();
+Console.WriteLine("Movies about baseball:");
+foreach (var movie in moviesAboutBaseball)
+{
+    Console.WriteLine($"  {movie}");
+}
+
+string complexQuery = @"
+[
+  {
+    $search: {
+      'compound': {
+        'must': [ {
+          'text': {
+             'query': ['Hawaii', 'Alaska'],
+             'path': 'plot'
+        },
+        {
+          'regex': {
+             'query': '([0-9]{4})',
+             'path': 'plot',
+             'allowAnalyzedField': true
+          }
+        } ],
+        'mustNot': [ {
+          'text': {
+            'query': ['Comedy', 'Romance'],
+            'path': 'genres'
+          }
+        },
+        {
+          'term': {
+            'query': ['Beach', 'Snow'],
+            'path': 'title'
+          }
+        } ]
+      }
+    }
+  },
+  {
+    $project: {
+      'title': 1,
+      'plot': 1,
+      'genres': 1,
+      '_id': 0
+    }
+  }
+]";
+
+var complexPipeline = Bson.Serialization.BsonSerializer
+	.Deserialize<BsonArray>(complexQuery)
```

Comment:
Deserializing a BsonArray is a neat trick but is unique to our JSON serializer. That may or may not be a reason to avoid it?

I suppose this example and the previous one should be implemented the same way (i.e. either start with an array of strings one per stage or use a single bigger JSON string for the entire array like you did here).

But doing them differently confuses me.

Reply:
Trick removed! :)